scalding-commons/src/main/java/com/twitter/scalding/commons/tap/VersionedTap.java

```diff
@@ -107,6 +108,21 @@ public void sinkConfInit(FlowProcess<JobConf> process, JobConf conf) {
   }
 
   @Override
+  public long getSize(JobConf conf) throws IOException {
+    Path path = new Path(getSourcePath(conf));
+    FileSystem fs = getFileSystem(conf);
+    FileStatus[] statuses = fs.globStatus(path);
```

Comment:
Does this get called after validateTaps? Hopefully otherwise there will be race conditions where this returns  0 but the file shows up later and then we estimate poorly